src/Common/src/TypeSystem/Common/InstantiatedType.cs

```diff
@@ -159,7 +159,22 @@ public override MethodDesc GetFinalizer()
             MethodDesc typicalFinalizer = _typeDef.GetFinalizer();
             if (typicalFinalizer == null)
                 return null;
-            return _typeDef.Context.GetMethodForInstantiatedType(typicalFinalizer, this);
+
+            MetadataType typeInHierarchy = this;
+            while (typicalFinalizer.OwningType.GetTypeDefinition() != typeInHierarchy.GetTypeDefinition())
```

Comment:
No it isn't. If the finalizer is implemented on a base type that is also a generic, then the _typeDef in that case is a MethodForInstantiatedType for an instantiated type which is instantiated over the open type variables of the derived type.